src/class-config.php

```diff
@@ -1188,613 +1200,423 @@ protected function add_field_group_fields( $field_group, $type_name, $layout = f
-						foreach ( $locations as $location ) {
-							if ( 'nav_menu' === $location['param'] && '!=' === $location['operator'] ) {
-								continue;
-							}
-							if ( 'nav_menu' === $location['param'] && '==' === $location['operator'] ) {
-								$menu_field_groups[] = $field_group;
-								break;
-							}
-						}
-					}
-				}
+
+			if ( ! isset( $field_group['graphql_types' ] ) ) {
+				continue;
 			}
-		}
 
-		if ( empty( $menu_field_groups ) ) {
-			return;
-		}
+			if ( empty( $field_group['graphql_types'] ) ) {
+				continue;
+			}
 
-		/**
-		 * Loop over the field groups for this post type
-		 */
-		foreach ( $menu_field_groups as $field_group ) {
+			if ( ! is_array( $field_group['graphql_types'] ) ) {
+				return;
+			}
 
-			$field_name = isset( $field_group['graphql_field_name'] ) ? $field_group['graphql_field_name'] : Config::camel_case( $field_group['title'] );
+			/**
+			 * Determine if the field group should be exposed
+			 * to graphql
+			 */
+			if ( ! $this->should_field_group_show_in_graphql( $field_group ) ) {
+				return;
+			}
 
+			$graphql_types = array_unique( $field_group['graphql_types'] );
+			$graphql_types = array_filter( $graphql_types );
+
+			/**
+			 * Prepare default info
+			 */
+			$field_name = isset( $field_group['graphql_field_name'] ) ? $field_group['graphql_field_name'] : Config::camel_case( $field_group['title'] );
 			$field_group['type'] = 'group';
 			$field_group['name'] = $field_name;
-			$description         = $field_group['description'] ? $field_group['description'] . ' | ' : '';
 			$config              = [
 				'name'            => $field_name,
-				'description'     => $description . sprintf( __( 'Added to the GraphQL Schema because the ACF Field Group "%s" was assigned to Menus', 'wp-graphql-acf' ), $field_group['title'] ),
 				'acf_field'       => $field_group,
 				'acf_field_group' => null,
-				'resolve'         => function( $root ) use ( $field_group ) {
+				'resolve'         => function ( $root ) use ( $field_group ) {
 					return isset( $root ) ? $root : null;
 				}
 			];
 
-			$this->register_graphql_field( 'Menu', $field_name, $config );
+			$qualifier =  sprintf( __( 'Added to the GraphQL Schema because the ACF Field Group "%1$s" was set to Show in GraphQL.', 'wp-graphql-acf' ), $field_group['title'] );
+			$config['description'] = $field_group['description'] ? $field_group['description'] . ' | ' . $qualifier : $qualifier;
 
+			/**
+			 * Loop over the GraphQL types for this field group on
+			 */
+			foreach ( $graphql_types as $graphql_type ) {
+				$this->register_graphql_field( $graphql_type, $field_name, $config );
+			}
 		}
-
 	}
 
 	/**
-	 * Add ACF Field Groups to Menu Items
+	 * Update field groups with graphql_types field
+	 * Used when upgrading version from under 0.4.1 to above
 	 *
-	 * @return void
+	 * @since 0.4.1
 	 */
-	protected function add_acf_fields_to_menu_items() {
+	public static function auto_update_field_groups() {
```

Comment:
This function attempts to update the database, but not all ACF field groups are saved in the database. 

Some ACF Field Groups are JSON files and some are PHP registered.

Instead of running a one-time script to update the database, we should instead check if the `graphql_types` field is set for the type, and if not, we should fallback to the old rules. 

I'm working on a PR to adjust this and account for field groups registered by JSON or PHP, in addition to field groups registered with the UI in the database.